### ISSUE TITLE
OTB fonts cause extremely wide window that cannot be resized

### ISSUE THREAD

**Issue by pkfbcedkrz** [lxqt/pcmanfm-qt#1515]
##### Expected Behavior
OTB fonts should not cause PcManFm-qt window to expand to width larger than screen, and horizontal resizing should still work.

##### Current Behavior
OTB fonts cause extremely wide window which cannot be resized horizontally.

##### Steps to Reproduce (for bugs)

1. Install any OTB font, such as Terminus (Debian: `xfonts-terminus`; Arch: `terminus-font`). Ensure that no other legacy version such as PCF is present. 
2. Set it as default qt5 font
3. Open PcManFm-qt

##### Context
Pango / HarfBuzz have dropped support for non-OTB bitmap fonts such as PCF and BDF. Going forward, the only option for bitmap fonts, which many users still prefer for sharper rendering at small sizes, will be OTB fonts.

Most distros have already begun converting legacy formats to OTB, as can be seen in the Debian and Arch repositories.

##### System Information

* PcManFm-qt version: All, including 0.17 and 1.0
* Distribution & Version: Debian Sid, Arch Linux
* Qt Version: 5.15.2
* liblxqt Version: 1.0

**Comment by tsujan** [comment]
pcmanfm-qt doesn't have a special font handling. The problem is in Terminus or how Qt handles it. Actually, it has problem with GTK too.

Terminus isn't for use in GUI but in terminals, editors, etc. Have you tried it with spacefm, Thunar, Dolphin or Nautilus? If not, please do.

**Comment by pkfbcedkrz** [comment]
@tsujan Yes, I have tried Terminus with various programs and desktop environments, as well as various other OTB fonts. pcmanfm-qt is the only program that displays this behavior - Dolphin and Deepin File Manager also do not, and neither does any other LXQt app.

Your comments reference problems with Terminus in GTK and elsewhere. What problems have you observed?

I use Terminus only as an example -- because it's a very popular bitmap terminal font, it was one of the first to be migrated from PCF (the previous standard for bitmap fonts) to OTB by distros like Debian, Arch, Fedora, etc. You can find more examples of OTB fonts here: https://gitlab.gnome.org/GNOME/pango/-/issues/534#note_1040472 . I have tested these fonts as above, and they all work as expected in GTK and other Qt apps. Please note that some font names might conflict with existing PCF or other legacy bitmap versions - fc-match should output an OTB font for a given name to make sure the right version is being used.

That being said, I understand this could an issue outside pcmanfm-qt. Since Dolphin and Deepin FM do not display this problem, perhaps it is a problem with libfm-qt? In any case, Qt apps still support PCF fonts, so OTB support is not as crucial as it is for GTK apps.

Thanks!

**Comment by tsujan** [comment]
Here, Terminus (`terminus-font` from Arch) creates invisible texts with Thunar, spacefm and Nautilus. With Dolphin, it's a disaster of another kind. I've also seen weird behaviors in some other Qt apps, which may have extremely wide windows with Terminus. These issues might be circumvented by a trick but their existence show that Terminus has a serious problem.

On the other hand, pcmanfm-qt or libfm-qt doesn't have a special font handling; hence closing the report.

That being said, I'm curious to know why Terminus creates these issues and what tricks can be used for avoiding them. Although I closed the report, I haven't closed my eyes to Terminus problems; will investigate them when I find the time.

**Comment by tsujan** [comment]
Thanks for your explanatory comment. But since I have Arch's `terminus-font`, it doesn't explain why you and I see different things with Thunar, spacefm and Dolphin, while we see the same thing with pcmanfm-qt.

**Comment by pkfbcedkrz** [comment]
Arch's terminus (`terminus-font`) works fine in Thunar here, but is invisible in Mousepad when italicized (eg, url inside comment in bash script). Debian's version (`fonts-terminus-otb`) has working italics:
[Uploading debian terminus.tar.gz…]()
 
On my Arch system, if I install `terminus-font` I see invisble text in italics. But if I add Debian's version to ~/.fonts, it overrides Arch's version and italics work properly in Mousepad. 

Not sure if that helps, but it seems Debian has used a better/more correct method to generate the OTBs.

**Comment by pkfbcedkrz** [comment]
More issues have been identified with Qt apps and OTB fonts: 

https://github.com/grassmunk/Chicago95/issues/218#issuecomment-1019075924

**Comment by jpka-** [comment]
Possible related: https://bugreports.qt.io/browse/QTBUG-82607
